### PR TITLE
docs(faq): add raw body option usage to faq

### PR DESCRIPTION
**Pull request by tolgap**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Docs
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #2310 

## What is the new behavior?

This PR adds a section in `FAQ` on how to enable the raw body option and how to use it on your controllers.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

The text is very barebones. It's just an initial commit to get things started. If necessary, we could add recipes on how to use this technique to validate Stripe or Shopify webhooks.

### CODE REVIEW

File: content/techniques/raw-body.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@Controller('cats')
+class CatsController {
+  @Post()
+  create(@Req() req: RawBodyRequest<Request>) {
+    const raw = req.rawBody; // returns a `Buffer`.
+  }
+}
+```
+
+#### Fastify example
```
Comment (Tony133) [inline content/techniques/raw-body.md:33]:
```suggestion
#### Use with Fastify
```

File: content/techniques/raw-body.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+### Raw body
+
+Nest applications have an option to provide the raw contents of the request body. It is disabled by default.
+
+One of the most common use-case for having access to the raw request body is performing webhook signature verifications. Usually to perform webhook signature validations the unserialized request body is required to calculate an HMAC hash.
+
+#### Express example
```
Comment (Tony133) [inline content/techniques/raw-body.md:7]:
```suggestion
#### Use with Express (default)
```

**Review by Tony133 — COMMENTED** [review]
I only reported two subtitle changes to align with the others that are in the documentation, then the rest is ok fine 👍

File: content/techniques/raw-body.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+To access the raw request body in a controller, a convenience interface `RawBodyRequest` is provided to expose a `rawBody` field on the request: use the interface `RawBodyRequest` type:
+
+```typescript
+import { Controller, Post, RawBodyRequest, Req } from '@nestjs/common';
+import { FastifyRequest } from 'fastify';
+
+@Controller('cats')
+class CatsController {
+  @Post()
+  create(@Req() req: RawBodyRequest<FastifyRequest>) {
```
Comment (micalevisk) [inline content/techniques/raw-body.md:57]: as another option, can't we use `@Req('rawBody')` while keeping that parameter type-safe? :thinking:

Comment (micalevisk) [reply content/techniques/raw-body.md:57]: maybe

```ts
@Req('rawBody') rawBody: RawBodyRequest['rawBody']
```

should work?

**Review by Tony133 — APPROVED** [review]

File: content/techniques/raw-body.md (hunk 3 above)
Comment (kamilmysliwiec) [reply content/techniques/raw-body.md:57]: I feel like `RawBodyRequest<FastifyRequest>` looks cleaner than `RawBodyRequest['rawBody']` but that's just my personal preference.

**Comment by kamilmysliwiec** [comment]
Very minor but can we move this under FAQ (from Techniques) category?

**Comment by tolgap** [comment]
@kamilmysliwiec moved it to FAQ.

File: content/faq/raw-body.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+### Raw body
+
+Nest applications have an option to provide the raw contents of the request body. It is disabled by default.
```
Comment (kamilmysliwiec) [inline content/faq/raw-body.md:3]:
```suggestion
```

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**